test/src/testPubSub.cpp

```diff
@@ -48,6 +48,20 @@ using namespace uprotocol::client;
 
 // capture and queue onReceive() for verification by the sender
 
+// The singleton reference shared by all tests has to be cleared on exit, or it causes a crash
+static std::shared_ptr<UpZenohClient> common_instance = nullptr;
+
+std::shared_ptr<UpZenohClient> getInstance()
+{
+    if (!common_instance) {
+        common_instance = UpZenohClient::instance(
+            BuildUAuthority().setName("device1").build(),
+            BuildUEntity().setName("rpc.client").setMajorVersion(1).setId(1).build());
+        EXPECT_NE(common_instance, nullptr);
+    }
+    return common_instance;
+}
+
```

Comment:
Is the intent to keep the same instance for every test case? I think it would be better to make an instance at the start of each test and clean it up at the end (using `SetUp()` and `TearDown()`).